data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/prepper/ServiceMapStatefulPrepper.java

```diff
@@ -155,24 +180,46 @@ private static File createPath(File path) {
                                 });
                     }
             ));
-        }));
-        try {
-            currentWindow.putAll(batchStateData);
-        } catch (RuntimeException e) {
-            LOG.error("Caught exception trying to put batch state data", e);
+        });
+    }
+
+    private void processSpan(final Span span, final Map<byte[], ServiceMapStateData> batchStateData) {
+        if (span.getServiceName() != null) {
+            final String serviceName = span.getServiceName();
+            final String spanId = span.getSpanId();
+            final String traceId = span.getTraceId();
+            final String parentSpanId = span.getParentSpanId();
+            try {
+                batchStateData.put(
+                        Hex.decodeHex(spanId),
+                        new ServiceMapStateData(
+                                serviceName,
+                                parentSpanId.isEmpty()? null : Hex.decodeHex(parentSpanId),
```

Comment:
It might make sense to create a method like `normalizeId` which then calls `decodeHex`. This way there is one function that normalizes an Id. But, since the logic is unlikely to change, I'll leave it up to you.

Reply:
Yeah. It is unlikely to change. Or we may even consider remove the Hex decode and use string as the key in MapDBPrepperState in the future.